set.mm

```diff
@@ -532760,6 +532537,78 @@ dependency is expressed in our hypothesis (called implicit
+    $( Infer one criterium for interchangeability from another.  (Contributed
+       by Wolf Lammen, 5-Aug-2023.) $)
+    wl-ich $p |- ( A. a A. b ( [ a / x ] [ b / y ] ph
+                              <-> [ b / x ] [ a / y ] ph )
+        -> ( ph <-> [ x / a ] [ y / x ] [ a / y ] ph ) ) $=
+      ( wsb wb wal wl-2spsbbi sbid2vw sbbii bitri sbco4 3bitr3g ) ACEFZBDFZACDF
+      ZBEFZGEHDHPDBFZECFZRDBFECFAQBCFDBFPRBCDEITOECFASOECODBJKAECJLABCDDEMN $.
+  $}
+
+  ${
+    $d b x $.
+    $( Lemma.  Simplify substitution expression when two particular variables
+       are the same.  (Contributed by Wolf Lammen, 5-Aug-2023.) $)
+    wl-ich-lem1 $p |- ( A. x x = y ->
+                         ( [ a / x ] [ b / y ] ph <-> [ b / x ] ph ) ) $=
+      ( weq wal wsb nfa1 drsb1 bicomd sbbid nfs1v sbf syl6bb ) BCFZBGZACEHZBDHA
+      BEHZBDHSQRSBDPBIQSRABCEJKLSBDABEMNO $.
+  $}
+
+  ${
+    $d a b x $.  $d a b y $.  $d a b ph $.
+    $( Lemma.  Factor out the common proof beginning for ~ wl-ich-lem4 and ?.
+       (Contributed by Wolf Lammen, 6-Aug-2023.) $)
+    wl-ich-lem2 $p |- ( A. x x = y
+             -> ( ( [ a / x ] [ b / y ] ph <-> [ b / x ] [ a / y ] ph )
+                      -> ( [ a / x ] ph <-> [ b / x ] ph ) ) ) $=
+      ( weq wal wsb wb wl-ich-lem1 bibi12d bicom1 syl6bi ) BCFBGZACEHBDHZACDHBE
+      HZIABEHZABDHZIRQINOQPRABCDEJABCEDJKQRLM $.
+  $}
+
+  ${
+    $d a b z $.  $d a b ph $.  $d a b ch $.
+    wl-ich-lem3.1 $e |- ( ph -> ( ps
+                          -> ( [ a / z ] ch <-> [ b / z ] ch ) ) ) $.
+    $( Lemma.  Factor out the common proof tail for ~ wl-ich-lem4 and ?.
+       (Contributed by Wolf Lammen, 6-Aug-2023.) $)
+    wl-ich-lem3 $p |- ( A. a A. b ps -> ( -. F/ z ch -> -. ph ) ) $=
+      ( wal wnf wsb wb 2alimdv sbnf2 syl6ibr com12 con3d ) BFHEHZACDIZAQRAQCDEJ
+      CDFJKZFHEHRABSEFGLCDEFMNOP $.
+  $}
+
+  ${
+    $d a b x $.  $d a b y $.  $d a b ph $.
+    $( Lemma.  To be meaningful, interchangeability should have ` x ` free in
+       ` ph ` .  This requires ` x ` and ` y ` be distinct, and the universe to
+       have more than one element.  Companion theorem is ~ wl-ich-lem5 .
+       (Contributed by Wolf Lammen, 5-Aug-2023.) $)
+    wl-ich-lem4 $p |- ( A. a A. b ( [ a / x ] [ b / y ] ph
+                                    <-> [ b / x ] [ a / y ] ph )
+                      -> ( -. F/ x ph -> -. A. x x = y ) ) $=
+      ( weq wal wsb wb wl-ich-lem2 wl-ich-lem3 ) BCFBGACEHBDHACDHBEHIABDEABCDEJ
+      K $.
+
+    $( Lemma.  To be meaningful, interchangeability should have ` y ` free in
+       ` ph ` .  This requires ` x ` and ` y ` be distinct, and the universe to
+       have more than one element.  Companion theorem is ~ wl-ich-lem5 .
+       (Contributed by Wolf Lammen, 6-Aug-2023.) $)
+    wl-ich-lem5 $p |- ( A. a A. b ( [ a / x ] [ b / y ] ph
+                                    <-> [ b / x ] [ a / y ] ph )
+                      -> ( -. F/ y ph -> -. A. x x = y ) ) $=
```

Comment:
For wl-ich-lem3/4/5, the positive forms (e.g., `( A. x x = y -> F/ y ph )`) seem more natural.

Reply:
I deliberately chose this form, because you usually (want to) have x free in ph, when addressing interchangeability.